src/devices/thermostat.cpp

```diff
@@ -2049,6 +2050,12 @@ bool Thermostat::set_wwmode(const char * value, const int8_t id) {
         }
         const uint8_t modes[] = {0, 5, 1, 2, 4};
         write_command(0x02F5 + dhw, 2, modes[set], 0x02F5 + dhw);
+    } else if (model() == EMSdevice::EMS_DEVICE_FLAG_R3000) { // Rego3000 - https://github.com/emsesp/EMS-ESP32/issues/1692
+        if (!Helpers::value2enum(value, set, FL_(enum_wwMode5))) {
+            return false;
+        }
+        const uint8_t modes[] = {1, 2, 0, 0, 5};
```

Comment:
This should be `{1, 2, 5}` for the 3 enum modes

We also need a remapping in `process_RC300WWmode`, line 1224 ff
```
    } else if (model() == EMSdevice::EMS_DEVICE_FLAG_R3000) {
        const uint8_t modes[] = {1, 2, 5}; //normal, comfort, eco+
        uint8_t       wwmode  = dhw->wwMode_ < sizeof(modes) ? modes[dhw->wwMode_] : EMS_VALUE_UINT8_NOTSET;
        telegram->read_value(wwmode, 2);
        const uint8_t modes1[] = {0, 0, 1, 0, 0, 2}; //0=normal (1), 1=comfort(2), 2=eco+(5)
        has_update(dhw->wwMode_, wwmode < sizeof(modes1) ? modes1[wwmode] : EMS_VALUE_UINT8_NOTSET);
``` 
mapping the unknown modes 0, 3, 4 to normal, or we set 
` modes1[] = {EMS_VALUE_UINT8_NOTSET, 0, 1, EMS_VALUE_UINT8_NOTSET, EMS_VALUE_UINT8_NOTSET, 2}`

Reply:
silly mistakes, thanks for checking. I'll make the changes, push the PR and then you can your bits. I see Tasmota have a new platform out for v3/IDF5 https://github.com/tasmota/platform-espressif32/releases/tag/2024.04.14